lib/deimos/consume/message_consumption.rb

```diff
@@ -28,12 +31,20 @@ def around_consume(payload, metadata)
       # Consume incoming messages.
       # @param _payload [String]
       # @param _metadata [Hash]
-      def consume(_payload, _metadata)
-        raise NotImplementedError
+      def consume(payload, metadata)
+        @klass.new.consume(payload, metadata)
       end
 
     private
 
+      def consumer_class(topic)
+        consumer_object = Deimos.config.consumer_objects.find do |object| 
+          object.topic == topic
+        end
+
+        consumer_object.class_name.constantize
+      end
```

Comment:
So rather than putting this in the consumer class, I think this belongs in the handler class. I see the handler class as being a very simple pass-through that just sorts the messages by topic and passes the messages or batches (depending on configuration) onto the appropriate consumer class. Consumer class shouldn't need to change at all in this PR.